server/services/discovery/tasks.py

```
@@ -356,33 +356,85 @@ def _get_sa_project_ids(token_data, user_id):
+        pid = p.get("projectId")
+        if not pid:
+            continue
+        try:
+            if has_active_billing(pid, credentials):
+                project_ids.append(pid)
+        except Exception:
+            project_ids.append(pid)
+    return project_ids
+
+
 def _get_oauth_project_ids(user_id):
```

Ruff (0.15.14)

[warning] Missing return type annotation for private function `_get_oauth_project_ids`

(ANN202)